hmda-data-publisher/src/test/scala/hmda/publisher/query/component/PublisherComponent2021Spec.scala

```diff
@@ -0,0 +1,131 @@
+package hmda.publisher.query.component
+
```

Comment:
One main general comment: while I shudder to think about adding more abstraction to anything in scala, has anyone thought about generalizing these tests (PublisherComponent, LarEntityImpl, possibly others?) As noted, the tests "pretty much work the same way." If it could be done, we wouldn't have to create and maintain separate tests for every year in perpetuity.

Reply:
I agree, there's a ton of refactoring that we can do on the current tests and beyond. A lot of data in the codebase in structured in a way that we have to "create and maintain separate X for every year" so that seems like a clean-up project during down-season.